File: iqchannels-sdk/src/main/java/ru/iqchannels/sdk/schema/UploadedFile.java
package ru.iqchannels.sdk.schema;


import androidx.annotation.Nullable;

/**
 * Created by Ivan Korobkov i.korobkov@iqstore.ru on 26/01/2017.
 */
public class UploadedFile {
    public String Id;
    public FileType Type;

    public FileOwnerType Owner;
    @Nullable public Long OwnerClientId;

    public ActorType Actor;
    @Nullable public Long ActorClientId;
    @Nullable public Long ActorUserId;

    public String Name; // Original file name.
    public String Path; // Relative filesystem path;
    public long Size; // Size in bytes.

    @Nullable public Integer ImageWidth;
    @Nullable public Integer ImageHeight;

    public String ContentType;
    public long CreatedAt;

    // Local
    @Nullable public String Url;
    @Nullable public String ImagePreviewUrl;
    @Nullable public String imageUrl;
}

File: iqchannels-sdk/src/main/java/ru/iqchannels/sdk/schema/UploadedFile.kt
package ru.iqchannels.sdk.schema

class UploadedFile {
	var Id: String? = null
	var Type: FileType? = null
	var Owner: FileOwnerType? = null
	var OwnerClientId: Long? = null
	var Actor: ActorType? = null
	var ActorClientId: Long? = null
	var ActorUserId: Long? = null
	var Name: String? = null // Original file name.
	var Path: String? = null // Relative filesystem path;
	var Size: Long = 0 // Size in bytes.
	var ImageWidth: Int? = null
	var ImageHeight: Int? = null
	var ContentType: String? = null
	var CreatedAt: Long = 0

	// Local
	var Url: String? = null
	var ImagePreviewUrl: String? = null
	var imageUrl: String? = null
}
